lib/models/finamp_models.dart

```diff
@@ -622,6 +696,441 @@ class DownloadedImage {
       );
 }
 
+/// A reference to a downloadable item with no state.  Can be freely created
+/// from a BaseItemDto at any time.  DownloadStubs/DownloadItems are considered
+/// equivalent if their types and ids match.
+@JsonSerializable(
+    fieldRename: FieldRename.pascal,
+    explicitToJson: true,
+    anyMap: true,
+    constructor: "_build")
+class DownloadStub {
+  DownloadStub._build({
+    required this.id,
+    required this.type,
+    required this.jsonItem,
+    required this.isarId,
+    required this.name,
+    required this.baseItemType,
+  }) {
+    assert(
+        _verifyEnums(), "$type $baseItemType ${baseItem?.toJson().toString()}");
+  }
+
+  bool _verifyEnums() {
+    switch (type) {
+      case DownloadItemType.collection:
+        return baseItem != null &&
+            BaseItemDtoType.fromItem(baseItem!) == baseItemType &&
+            baseItemType != BaseItemDtoType.song &&
+            baseItemType != BaseItemDtoType.unknown;
+      case DownloadItemType.song:
+        return baseItemType == BaseItemDtoType.song &&
+            baseItem != null &&
+            BaseItemDtoType.fromItem(baseItem!) == baseItemType;
+      case DownloadItemType.image:
+        return baseItem != null;
+      case DownloadItemType.finampCollection:
+        // TODO create an enum or somthing for this if more custom collections happen
```

Comment:
Yeah, that's mostly why I never ended up changing this instead of trying to get it cleaner.  I don't know dynamic they might really be or how they get represented in code.